configuration/scripts/cmake/CMakeLists.txt

```
@@ -0,0 +1,379 @@
+  ${ICEPACK}/columnphysics/icepack_flux.F90
+  ${ICEPACK}/columnphysics/icepack_fsd.F90
+  ${ICEPACK}/columnphysics/icepack_intfc.F90
+  ${ICEPACK}/columnphysics/icepack_isotope.F90
+  ${ICEPACK}/columnphysics/icepack_itd.F90
+  ${ICEPACK}/columnphysics/icepack_kinds.F90
+  ${ICEPACK}/columnphysics/icepack_mechred.F90
+  ${ICEPACK}/columnphysics/icepack_meltpond_lvl.F90
+  ${ICEPACK}/columnphysics/icepack_meltpond_topo.F90
+  ${ICEPACK}/columnphysics/icepack_mushy_physics.F90
+  ${ICEPACK}/columnphysics/icepack_ocean.F90
+  ${ICEPACK}/columnphysics/icepack_orbital.F90
+  ${ICEPACK}/columnphysics/icepack_parameters.F90
+  ${ICEPACK}/columnphysics/icepack_shortwave_data.F90
+  ${ICEPACK}/columnphysics/icepack_shortwave.F90
+  ${ICEPACK}/columnphysics/icepack_snow.F90
+  ${ICEPACK}/columnphysics/icepack_therm_bl99.F90
+  ${ICEPACK}/columnphysics/icepack_therm_itd.F90
+  ${ICEPACK}/columnphysics/icepack_therm_mushy.F90
+  ${ICEPACK}/columnphysics/icepack_therm_shared.F90
+  ${ICEPACK}/columnphysics/icepack_therm_vertical.F90
+  ${ICEPACK}/columnphysics/icepack_tracers.F90
+  ${ICEPACK}/columnphysics/icepack_warnings.F90
+  ${ICEPACK}/columnphysics/icepack_wavefracspec.F90
+  ${ICEPACK}/columnphysics/icepack_zbgc.F90
+  ${ICEPACK}/columnphysics/icepack_zbgc_shared.F90
+
+  # Shared C
+  ${CICE_CORE}/cicedyn/infrastructure/ice_shr_reprosum86.c
+
+  # MPI
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_boundary.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_broadcast.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_communicate.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_exit.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_gather_scatter.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_global_reductions.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_reprosum.F90
+  ${CICE_CORE}/cicedyn/infrastructure/comm/mpi/ice_timers.F90
+)
+
+if(CICE_DRIVER MATCHES "nuopc/cmeps")
+  target_sources(cicelib PRIVATE
+    # NUOPC CMEPS driver
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/CICE_FinalMod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/CICE_InitMod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/CICE_RunMod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/cice_wrapper_mod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/ice_comp_nuopc.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/ice_import_export.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/ice_mesh_mod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/ice_prescribed_mod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/ice_scam.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/ice_shr_methods.F90
+  )
+elseif(CICE_DRIVER MATCHES "standalone/cice")
+  target_sources(cicelib PRIVATE
+    # CICE standalone
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/CICE_FinalMod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/CICE_InitMod.F90
+    ${CICE_CORE}/drivers/${CICE_DRIVER}/CICE_RunMod.F90
+  )
+else()
+  message(FATAL_ERROR "CICE_DRIVER: ${CICE_DRIVER} not supported, add to cmake")
```

Not sure if the error message will be very clear to users. Maybe the following would be better?

```suggestion
  message(FATAL_ERROR "CICE_DRIVER: ${CICE_DRIVER} is currently not supported by the CMake build system")
```